gui-lib/framework/private/text-port.rkt

```diff
@@ -290,7 +290,10 @@
         (channel-put box-read-chan (cons eof (position->line-col-pos unread-start-point))))
       (define/public-final (clear-input-port) (channel-put clear-input-chan (void)))
       (define/public-final (clear-box-input-port) (channel-put box-clear-input-chan (void)))
-      (define/public-final (clear-output-ports) 
+
+      (define output-ports-clear-count 0)
+      (define/public-final (clear-output-ports)
+        (set! output-ports-clear-count (add1 output-ports-clear-count))
```

Comment:
@rfindler this is the only place that writes to `output-ports-clear-count`, so provided that there is only one thread that calls `(clear-output-ports)`, I think there won't be a race condition?